server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionTaskManager.java

```diff
@@ -283,7 +272,7 @@ public synchronized void removeRunningTaskFromList(AbstractCompactionTask task)
    *
    * @throws RejectedExecutionException
    */
-  public synchronized void submitTask(
+  public void submitTask(
       String fullStorageGroupName, long timePartition, Callable<Void> compactionMergeTask)
       throws RejectedExecutionException {
     if (taskExecutionPool != null && !taskExecutionPool.isTerminated()) {
```

Comment:
If this condition is `false`, the sub-thread won't be started, which will lead to the `semaphore` cannot be `released`. We need to handle the scenario here